cost/azure/databricks/rightsize_compute/azure_databricks_rightsize_compute.pt

```diff
@@ -0,0 +1,1478 @@
+  result "result"
+  code <<-EOS
+  underutil_total_savings = 0.0
+  idle_total_savings = 0.0
+
+  underutil_list = []
+  idle_list = []
+
+  _.each(ds_filtered_workspace_virtualmachines_with_costs, function(vm) {
+    // Get the utilization metrics that we are going to use for comparison
+    cpu_value = vm.metricsForPeriod["cpu_"+param_stats_threshold.toLowerCase()]
+    mem_value = vm.metricsForPeriod["memory_"+param_stats_threshold.toLowerCase()+"_percent"]
+
+    // Set thresholdType for output
+    vm["thresholdType"] = param_stats_threshold
+    vm["thresholdResourceTypes"] = param_stats_check_both
+
+    // Test for whether to consider the instance idle or underutilized.
+    // Assume instance is not idle or underutilized by default.
+    is_idle = false
+    is_underutil = false
+
+    // Determine if the instance is idle or underutilized for each category.
+    // Store boolean result for later use.
+    is_idle_cpu = cpu_value < param_stats_idle_threshold_cpu_value
+    is_underutil_cpu = cpu_value < param_stats_underutil_threshold_cpu_value
+    is_idle_mem = mem_value < param_stats_idle_threshold_mem_value
+    is_underutil_mem = mem_value < param_stats_underutil_threshold_mem_value
+
+      // Determine whether we're checking for CPU, memory, or both
+    checking_cpu = param_stats_underutil_threshold_cpu_value != -1 || param_stats_idle_threshold_cpu_value != -1
+    checking_mem = param_stats_underutil_threshold_mem_value != -1 || param_stats_idle_threshold_mem_value != -1
+
+    // If we're only checking CPU, simply set is_idle/is_underutil to their CPU equivalents
+    if (!checking_mem) { is_idle = is_idle_cpu }
+    if (!checking_mem) { is_underutil = is_underutil_cpu }
+
+    // If we're only checking memory, simply set is_idle/is_underutil to their memory equivalents
+    if (!checking_cpu) { is_idle = is_idle_mem }
+    if (!checking_cpu) { is_underutil = is_underutil_mem }
+
+    // If we're checking both, do an 'and' or an 'or' depending on the value of param_stats_check_both
+    if (checking_cpu && checking_mem) {
+      if (param_stats_check_both == "Both CPU and Memory") {
+        is_idle = is_idle_cpu && is_idle_mem
+        is_underutil = is_underutil_cpu && is_underutil_mem
+      } else {
+        is_idle = is_idle_cpu || is_idle_mem
+        is_underutil = is_underutil_cpu || is_underutil_mem
+      }
+    }
+
+    // Only bother doing anything if we're checking at least one metric
+    if (checking_cpu || checking_mem) {
+      if (is_idle) { // First check to see if the resource is below the lower "idle" thresholds
+        vm["recommendationType"] = "Delete"
+        vm["recommendationDetails"] = [
+          "Delete Azure virtual machine ", vm["resourceId"], " ",
```

Comment:
Might be worth modifying this text to be specific that this is a Databricks recommendation

Reply:
Updated in 1eb5c1ee434d05a2368db673801ff4d40c7bea0c
```javascript
"Resize Databricks Cluster "+vm["databricks_cluster_id"]+" which created Azure virtual machine ", vm["resourceId"], " ",
          "in Azure Subscription ", vm["subscriptionName"], " ",
...
```

Reply:
Example Summary now:
> Resize Databricks Cluster 0921-182737-cblrjym3 which created Azure virtual machine /subscriptions/0db6431e-7324-4088-aae6-772079b7e198/resourcegroups/bk-databricks-test-managed-resource-group/providers/microsoft.compute/virtualmachines/616a5a227b7b4e0c859db96b2737cd47 in Azure Subscription  (0db6431e-7324-4088-aae6-772079b7e198) from Standard_DS3_v2 to Standard_DS2_v2